### PR TITLE
Add UTXO APIs

### PR DESCRIPTION
**Pull request by jdkanani**
- Add UTXO APIs. Store UTXOs in DB, indexed by address, block number, tx index, and output index
- Prevent duplicate transactions in tx pool
- Prevent spent transaction in `sendTx` API

### CODE REVIEW

File: src/server/routes.js [hunk 1]
```diff
@@ -47,9 +48,28 @@ const methods = {
...
+  },
+
+  async getTxByHash([txHash]) {
+    let tx = {}
+    try {
+      const obj = await getTxByHash(txHash)
+      if (obj) {
+        tx = obj.toJSON(true)
+      }
+    } catch (e) {}
```
Comment (federicobond) [inline src/server/routes.js]: These catch-all exception handlers can mask potential errors

File: src/chain/transaction.js [hunk 2]
```diff
@@ -208,6 +211,25 @@ export default class Transaction {
...
+  getSender2() {
+    return this._getSender(1)
+  }
+
+  _getSender(i) {
+    if (this._inputNull(i)) {
+      return null
+    }
+
+    const vrs = utils.fromRpcSig(this.raw[11 + i]) // parse {v,r,s} from sig
```
Comment (federicobond) [inline src/chain/transaction.js]: These magic numbers make code hard to follow. Can we define properties to access each part perhaps?

**Review by federicobond — COMMENTED** [review]
LGTM, I made some minor comments.

File: src/server/routes.js (hunk 1 above)
Comment (jdkanani) [reply src/server/routes.js]: Yeah, I'll remove them.

File: src/chain/transaction.js (hunk 2 above)
Comment (jdkanani) [reply src/chain/transaction.js]: Will fix them.

**Comment by jdkanani** [comment]
Pushed utility functions for clarity.

**Review by federicobond — APPROVED** [review]
LGTM
